From 5a268edc0a2e196e2f489c628779a3bdfe52368e Mon Sep 17 00:00:00 2001
From: Eduardo Ferreira <eduardo.barbosa@toradex.com>
Date: Thu, 20 Jun 2024 09:15:52 -0300
Subject: [PATCH] daemon: use default system config when none is available

This allows the system image to provide a default daemon.json file while
still allowing the user to overwrite via /etc/docker/daemon.json.

Upstream-Status: Inappropriate [embedded specific]
Signed-off-by: Ricardo Salveti <ricardo@foundries.io>
Signed-off-by: Sergio Prado <sergio.prado@toradex.com>
Signed-off-by: Eduardo Ferreira <eduardo.barbosa@toradex.com>
Signed-off-by: Hiago De Franco <hiago.franco@toradex.com>
---
 daemon/command/daemon.go | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/daemon/command/daemon.go b/daemon/command/daemon.go
index 688ddf085df0..7892f8a8cdcf 100644
--- a/daemon/command/daemon.go
+++ b/daemon/command/daemon.go
@@ -571,6 +571,12 @@ func loadDaemonCliConfig(opts *daemonOptions) (*config.Config, error) {
 	}
 	opts.setDefaultOptions()
 
+  // UNIX: use default system daemon config file if provided is not available
+  defaultSystemDaemonConfigFile := "/usr/lib/docker/daemon.json"
+  if _, err := os.Stat(opts.configFile); os.IsNotExist(err) {
+    opts.configFile = defaultSystemDaemonConfigFile
+  }
+
 	conf := opts.daemonConfig
 	flags := opts.flags
 	conf.Debug = opts.Debug
-- 
2.51.0

